### PR TITLE
Data loading

### PR DESCRIPTION
**Pull request by jacktrig**
many changes to get data loading to work for admin json, students loading started, commented out code that doesn't compile

### CODE REVIEW

**Comment by josephaw1022** [comment]
# We're not really on the same page with how we are handling the data
 
- It unnecessary to remove the methods from the endpoints and implement data handling methods in non-endpoint methods. 
- This completely throws out the separation of concerns and makes the code more confusing
- Instead of just turning the endpoints zombie code and doing what we want in the endpoints in another class, why not just implement the data fetching logic into the endpoint class 
- Just implement the endpoints as the  singleton instance and stop handling the data in a non-endpoint class

**Comment by josephaw1022** [comment]
Merge Conflict & pulling 
-- 

Also, make sure you pull from master next time (if you didn't) before making the branch because there's literal merge conflict characters left in the code, which is big no-no.

**Comment by jacktrig** [comment]
I commented out all the code in the endpoints because it doesn't compile. You can't reference "this" from a static method.  The data handling could possibly be moved to the endpoint later but following portias examples it was easier to make separate classes while understanding how the data loading and parsing will work.  

You can see the conflict characters on the commit history but not in the actual code.

**Comment by josephaw1022** [comment]
# Put the logic you just made in the endpoints

- It's easier to simply edit the methods types and or return types on the endpoints 
- I don't care if you change the method or return types or fix the syntax in the endpoint classes, but we are not going to start making our code unorganized and have classes with "duplicate purposes."  We all know when we say maybe later, that it won't get done. 
- Why don't you simply implement the same logic in the endpoints and change the types? That's far better than rewriting what we already have. 
- The purpose of endpoints is so that we don't have to remember the specific methods for each and every entity such as student, user, company, etc...
- Just add a private constructor on the endpoints and load the data from the json in the endpoint constructor.
- Also, add another method that saves the data from the list attribute in the endpoint and saves it into the json

**Comment by josephaw1022** [comment]
Look in the files changed tab. The merge conflicts are actually in the code 
-- 

Screenshot (15)

File: src/DataWriter.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ 
+            file.write(adminsJSON.toJSONString());
+            file.flush();
+ 
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+	}
+	
+	public static JSONObject getAdminJSON (Admin admin) {
```
Comment (josephaw1022) [inline src/DataWriter.java:30]: Should be done in endpoint constructor

File: src/DataWriter.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ 
+            file.write(studentsJSON.toJSONString());
+            file.flush();
+ 
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+	}
+	
+	public static JSONObject getStudentJSON (Student student) {
```
Comment (josephaw1022) [inline src/DataWriter.java:64]: should be done in student endpoint

**Comment by jacktrig** [comment]
This needs to be merged so the group can pull the changes I've made to the base classes in order to implement data loading.

**Pull request merged**